[imgui] build failure

**Host Environment**
- OS: Windows
- Compiler: Ninja

**To Reproduce**
Steps to reproduce the behavior:
`vcpkg install imgui[core,win32-binding,dx12-binding,dx11-binding] --recurse`

**Failure logs**
```
C:\Windows\System32>vcpkg install imgui[core,win32-binding,dx12-binding,dx11-binding] --recurse
Computing installation plan...
The following packages will be built and installed:
    imgui[core,dx11-binding,dx12-binding,win32-binding]:x86-windows -> 1.87
Detecting compiler hash for triplet x86-windows...
Restored 0 packages from C:\Users\low\AppData\Local\vcpkg\archives in 116.5 us. Use --debug to see more details.
Starting package 1/1: imgui:x86-windows
Building package imgui[core,dx11-binding,dx12-binding,win32-binding]:x86-windows...
-- Note: imgui only supports static library linkage. Building static library.
-- Using cached ocornut-imgui-v1.87.tar.gz.
-- Cleaning sources at C:/vcpkg/buildtrees/imgui/src/v1.87-1654d039d7.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/ocornut-imgui-v1.87.tar.gz
-- Using source at C:/vcpkg/buildtrees/imgui/src/v1.87-1654d039d7.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: C:/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j9
    Working Directory: C:/vcpkg/buildtrees/imgui/x86-windows-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\imgui\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:112 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/imgui/portfile.cmake:70 (vcpkg_cmake_install)
  scripts/ports.cmake:146 (include)


error: building imgui:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+imgui
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[imgui]+Build+error
including:
  package: imgui[core,dx11-binding,dx12-binding,win32-binding]:x86-windows -> 1.87
    vcpkg-tool version: 2022-03-30-692785ac944e81417840c6de244fb3e18a4b35eb
    vcpkg-scripts version: 0d7603c4e 2022-05-09 (3 weeks ago)

Additionally, attach any relevant sections from the log files above.
```
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/8790754/install-x86-windows-dbg-out.log)

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.
I tried searching Google, nothing. I tried choosing different packages. Same error. However when I installed only core and win32-binding it worked. But I need DirectX now too.

--- a/ports/imgui/portfile.cmake
+++ b/ports/imgui/portfile.cmake
@@ -21,10 +21,6 @@ endif()
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/imgui-config.cmake.in" DESTINATION "${SOURCE_PATH}")
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
 
-if(("metal-binding" IN_LIST FEATURES OR "osx-binding" IN_LIST FEATURES) AND (NOT VCPKG_TARGET_IS_OSX))
-    message(FATAL_ERROR "Feature metal-binding and osx-binding are only supported on osx.")
-endif()
-
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES 
     allegro5-binding            IMGUI_BUILD_ALLEGRO5_BINDING

--- a/ports/imgui/vcpkg.json
+++ b/ports/imgui/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "imgui",
   "version": "1.87",
+  "port-version": 1,
   "description": "Bloat-free Immediate Mode Graphical User interface for C++ with minimal dependencies.",
   "homepage": "https://github.com/ocornut/imgui",
   "license": "MIT",
@@ -25,16 +26,20 @@
       "description": "Build with docking support"
     },
     "dx10-binding": {
-      "description": "Make available DirectX10 binding"
+      "description": "Make available DirectX10 binding",
+      "supports": "windows & !uwp"
     },
     "dx11-binding": {
-      "description": "Make available DirectX11 binding"
+      "description": "Make available DirectX11 binding",
+      "supports": "windows & !uwp"
     },
     "dx12-binding": {
-      "description": "Make available DirectX12 binding"
+      "description": "Make available DirectX12 binding",
+      "supports": "!x86 & windows & !uwp"
     },
     "dx9-binding": {
-      "description": "Make available DirectX9 binding"
+      "description": "Make available DirectX9 binding",
+      "supports": "windows & !uwp"
     },
     "freetype": {
       "description": "Build font atlases using FreeType instead of stb_truetype",
@@ -58,7 +63,8 @@
       "description": "Install the libigl-imgui headers"
     },
     "metal-binding": {
-      "description": "Make available Metal binding"
+      "description": "Make available Metal binding",
+      "supports": "osx"
     },
     "opengl2-binding": {
       "description": "Make available OpenGL (legacy) binding"
@@ -67,7 +73,8 @@
       "description": "Make available OpenGL3/ES/ES2 (modern) binding"
     },
     "osx-binding": {
-      "description": "Make available OSX binding"
+      "description": "Make available OSX binding",
+      "supports": "osx"
     },
     "sdl2-binding": {
       "description": "Make available SDL2 binding",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2930,7 +2930,7 @@
     },
     "imgui": {
       "baseline": "1.87",
-      "port-version": 0
+      "port-version": 1
     },
     "imgui-sfml": {
       "baseline": "2.5",

--- a/versions/i-/imgui.json
+++ b/versions/i-/imgui.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "647accc743e0af0547fbf8671c170038e202fd1e",
+      "version": "1.87",
+      "port-version": 1
+    },
+    {
       "git-tree": "b9a40068f3905099cc5cb247f1b488d0e27dfd7f",
       "version": "1.87",
       "port-version": 0

